catroid/src/main/java/org/catrobat/catroid/embroidery/EmbroideryListManager.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.embroidery;
+
+import android.graphics.PointF;
+
+import org.catrobat.catroid.content.Sprite;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class EmbroideryListManager {
```

Comment:
wouldnt it be easier to just have the layer set in the stitchpoint, instead of all this map-ception?

you could write a comparator to sort, or filter for stitchpoints based on their layer, and this way round create the dst stitchblocks.

also, without having a deeper look into it, i think using the name of the sprite as identification will probably break when you use clones.